spacy/tokens/underscore.py

```diff
@@ -88,8 +95,46 @@ def get(self, name: str) -> Any:
     def has(self, name: str) -> bool:
         return name in self._extensions
 
-    def _get_key(self, name: str) -> Tuple[str, str, Optional[int], Optional[int]]:
-        return ("._.", name, self._start, self._end)
+    def _get_key(
+        self, name: str
+    ) -> Union[
+        Tuple[str, str, Optional[int], Optional[int]],
+        Tuple[
+            str,
+            str,
+            Optional[int],
+            Optional[int],
+            Optional[Union[str, int]],
+            Optional[Union[str, int]],
+            Optional[Union[str, int]],
+        ],
+    ]:
+        if hasattr(self, "_label"):
+            return (
+                "._.",
+                name,
+                self._start,
+                self._end,
+                self._label,
+                self._kb_id,
+                self._span_id,
+            )
+        else:
+            return ("._.", name, self._start, self._end)
+
+    @staticmethod
+    def _replace_keys(old_underscore, new_underscore):
+        """
+        This function is called by Span when its kb_id or label are re-assigned.
+        It checks if any user_data is stored for this span and replaces the keys
+        """
+        for name in old_underscore._extensions:
+            old_key = old_underscore._get_key(name)
+            new_key = new_underscore._get_key(name)
+            if old_key in old_underscore._doc.user_data and old_key != new_key:
```

Comment:
Do the `old_key != new_key` comparison first in this `if` statement, as it's going to be quicker than the `in` statement, and as soon as the first condition is `False`, the second one won't be evaluated anymore.